stripe/src/test/java/com/stripe/android/view/ExpiryDateEditTextTest.java

```diff
@@ -57,6 +57,40 @@ public void afterInputTwoDigits_textContainsSlash() {
     }
 
     @Test
+    public void inputSingleDigit_whenDigitIsTooLargeForMonth_prependsZero() {
+        mExpiryDateEditText.append("4");
+        assertEquals("04/", mExpiryDateEditText.getText().toString());
+        assertEquals(3, mExpiryDateEditText.getSelectionStart());
+    }
+
+    @Test
+    public void inputSingleDigit_whenDigitIsZeroOrOne_doesNotPrependZero() {
+        mExpiryDateEditText.append("1");
+        assertEquals("1", mExpiryDateEditText.getText().toString());
+        assertEquals(1, mExpiryDateEditText.getSelectionStart());
+    }
+
+    @Test
+    public void inputSingleDigit_whenAtFirstCharacterButTextNotEmpty_doesNotPrependZero() {
+        mExpiryDateEditText.append("1");
+        mExpiryDateEditText.setSelection(0);
+        mExpiryDateEditText.getEditableText().replace(0, 0, "3", 0, 1);
+
+        assertEquals("31/", mExpiryDateEditText.getText().toString());
+        assertEquals(1, mExpiryDateEditText.getSelectionStart());
+    }
+
+    @Test
+    public void inputMultipleDigits_whenStartingFromEmpty_doesNotPrependZero() {
+        // This isn't a valid date, but we don't want to change complicated pastes.
+        mExpiryDateEditText.append("23");
```

Comment:
I don't think a lot of people are going to be pasting in dates on mobile, but I don't want the control to freak out.